src/core/constraint.jl

```diff
@@ -14,99 +14,140 @@ end
 
 # Generic thermal limit constraint
 "`p[f_idx]^2 + q[f_idx]^2 <= rate_a^2`"
-function constraint_thermal_limit_from(pm::AbstractPowerModel, n::Int, f_idx, rate_a)
+function constraint_thermal_limit_from(pm::AbstractPowerModel, n::Int, f_idx, rate_a; constraint_name=nothing)
     p_fr = var(pm, n, :p, f_idx)
     q_fr = var(pm, n, :q, f_idx)
 
-    JuMP.@constraint(pm.model, p_fr^2 + q_fr^2 <= rate_a^2)
+    #Generate default name if none is provided
+    constraint_name = isnothing(constraint_name) ? "thermal_limit_from[$f_idx]" : constraint_name
+
+    JuMP.@constraint(pm.model, Symbol(constraint_name), p_fr^2 + q_fr^2 <= rate_a^2)

```

Comment:
This won't work. The `@constraint` macro requires an identifier. You would need to either interpolate a symbol or register the name manually (cleaner and easier).

```suggestion
    c = JuMP.@constraint(pm.model, p_fr^2 + q_fr^2 <= rate_a^2, base_name = constraint_name)
    
    pm.model[Symbol(constraint_name)] = c
```

Reply:
Done